addons/api/addon/serializers/host-catalog.js

```diff
@@ -0,0 +1,41 @@
+import ApplicationSerializer from './application';
+
+export default class HostCatalogSerializer extends ApplicationSerializer {
```

Comment:
I think we do need it.

I tested following next steps:
1. Serializer returns `serialized` when the record is `static` type (rather than calling `serializeStatic`).
2. Static unit test creates a record as follows, and expect this as result. 
```javascript
{
  name: 'static',
  description: 'this is a static host-catalog',
  type: 'static',
}
```
3. Test fails because it received:
```javascript
{
  attributes: {
    client_id: null,
    disable_credential_rotation: false,
    region: null,
    subscription_id: null,
    tenant_id: null
  },
  description: "this is a static host-catalog",
  name: "static",
  type: "static"
}
```
4. As per my understanding, we should receive `attributes` but we are, that's why I add the `delete.attributes` within the `serializeStatic`

If you think I am doing something wrong or missing something, happy to jump on a zoom call.

Reply:
Same behaviour with `serializeDynamic`. If is an aws host-catalog brings (set to null) fields from azure, that's why I delete them.